src/components/ListItem.jsx

```diff
@@ -1,5 +1,30 @@
+import { updateItem } from '../api';
+import { subtractDates } from '../utils';
 import './ListItem.css';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ name, listPath, id, dateLastPurchased }) {
+	const todaysDate = new Date();
+
+	const handleChecked = async () => {
+		try {
+			await updateItem(listPath, id, todaysDate);
+		} catch (err) {
+			console.error(err);
+		}
+	};
+
+	return (
+		<li className="ListItem">
+			<label>
+				{name}
+				<input
+					type="checkbox"
+					id={`checkbox-${id}`} // Unique identifier
+					name={name}
+					onChange={handleChecked}
+					checked={subtractDates(todaysDate, dateLastPurchased)}
```

Comment:
I wonder if updating this line to 

```javascript

checked={Timestamp.fromDate(getPastDate(1)) < dateLastPurchased}

```

And then adding a new `getPastDate` function to `src/utils/dates` similar to the existing `getFutureDate` with a `-` instead of a `+` might work?

```javascript

export function getPastDate(offset) {
	return new Date(Date.now() - offset * ONE_DAY_IN_MILLISECONDS);
}

```

Then you'd have two nice reusable utility functions for that future sorting by date issue. Totally up for other solutions too if you're not into the idea.

Reply:
@StefieCaff thanks for looking into this last night! I think I came up with a solution based on @deeheber great feedback/suggestion. Let me know what you think/if it works for you. It's implementing `Timestamp` and cleaning up that `convertTimestamp` function! I also changed `key={item.name}` to `key={item.id}` in `List` because I was getting a duplicate key error (I had two items with the same name) and it was driving me crazy 🤪

Reply:
Looks great, thank you both for buttoning this up. The offset/(1000) is what I was missing last night, but I couldn't see it. I am going to keep a mental note about the `getPastDate` function above 😄.

🐜 Good call, the` key` def, needed to be updated, I'll double check the any other lists...(do we have any other lists 🤷🏻)... I also think it would be a good idea to let the rest of the team know.